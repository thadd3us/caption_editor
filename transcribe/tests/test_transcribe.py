"""Tests for transcription tool."""

import subprocess
from pathlib import Path


def test_transcribe_osr_audio(repo_root: Path, tmp_path: Path, snapshot):
    """Test transcribing the OSR audio file with 10s chunks and 5s overlap."""
    test_audio = repo_root / "tests" / "fixtures" / "OSR_us_000_0010_8k.wav"
    output_path = tmp_path / "output.vtt"

    # Path to transcribe script
    transcribe_script = repo_root / "transcribe" / "transcribe.py"

    # Run transcription
    result = subprocess.run(
        [
            "python3",
            str(transcribe_script),
            str(test_audio),
            "--output",
            str(output_path),
            "--chunk-size",
            "10",
            "--overlap",
            "5",
        ],
        capture_output=True,
        text=True,
    )

    # Check that command succeeded
    assert result.returncode == 0, f"Transcription failed: {result.stderr}"

    # Check that output file was created
    assert output_path.exists(), "Output VTT file was not created"

    # Read and snapshot the VTT content
    vtt_content = output_path.read_text()

    # Use syrupy to snapshot the output
    assert vtt_content == snapshot
