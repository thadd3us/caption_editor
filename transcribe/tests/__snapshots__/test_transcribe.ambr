# serializer version: 1
# name: test_transcribe_osr_audio[nvidia/parakeet-tdt-0.6b-v3]
  '''
  WEBVTT
  
  NOTE {"id": "2ea43707-088b-c4fe-c7ff-b59f4a1232a0", "mediaFilePath": "/code/tests/fixtures/OSR_us_000_0010_8k.wav"}
  
  
  NOTE {"id": "ccdb12d8-0423-3909-3977-c4fc906f2122", "timestamp": "2025-10-29T18:32:06.553163+00:00"}
  
  ccdb12d8-0423-3909-3977-c4fc906f2122
  00:00:00.240 --> 00:00:03.440
  The birch canoe slid on the smooth planks.
  
  
  NOTE {"id": "57c15f3a-9d2c-f253-abdc-2554cec2381d", "timestamp": "2025-10-29T18:32:06.553163+00:00"}
  
  57c15f3a-9d2c-f253-abdc-2554cec2381d
  00:00:04.000 --> 00:00:07.200
  Glue the sheet to the dark blue background.
  
  
  NOTE {"id": "1566b07d-6382-fe9f-9640-6107ea4c603c", "timestamp": "2025-10-29T18:32:06.553163+00:00"}
  
  1566b07d-6382-fe9f-9640-6107ea4c603c
  00:00:07.440 --> 00:00:09.760
  It is easy to tell the depth of a well.
  
  
  NOTE {"id": "2942d461-d0a7-7108-8341-56bb79409b0a", "timestamp": "2025-10-29T18:32:06.553163+00:00"}
  
  2942d461-d0a7-7108-8341-56bb79409b0a
  00:00:10.480 --> 00:00:13.520
  These days a chicken leg is a rare dish.
  
  
  NOTE {"id": "1d190074-17ce-7bb5-c63a-d1bf50b687cb", "timestamp": "2025-10-29T18:32:06.553163+00:00"}
  
  1d190074-17ce-7bb5-c63a-d1bf50b687cb
  00:00:14.000 --> 00:00:16.720
  Rice is often served in round bowls.
  
  
  NOTE {"id": "0a3f524e-beec-08e3-5769-12b2edc3d8ec", "timestamp": "2025-10-29T18:32:06.553163+00:00"}
  
  0a3f524e-beec-08e3-5769-12b2edc3d8ec
  00:00:17.000 --> 00:00:19.800
  The juice of lemons makes fine punch.
  
  
  NOTE {"id": "5f807269-18e2-5659-1703-8f5bb1be6356", "timestamp": "2025-10-29T18:32:06.553163+00:00"}
  
  5f807269-18e2-5659-1703-8f5bb1be6356
  00:00:20.240 --> 00:00:22.880
  The box was thrown beside the parked truck.
  
  
  NOTE {"id": "04fb3db7-8763-bb91-a17f-0107cb7f4dcb", "timestamp": "2025-10-29T18:32:06.553163+00:00"}
  
  04fb3db7-8763-bb91-a17f-0107cb7f4dcb
  00:00:23.440 --> 00:00:26.240
  The hogs were fed chopped corn and garbage.
  
  
  NOTE {"id": "17184080-9b9d-983c-8a5f-d8df57cc702a", "timestamp": "2025-10-29T18:32:06.553163+00:00"}
  
  17184080-9b9d-983c-8a5f-d8df57cc702a
  00:00:26.800 --> 00:00:29.280
  Four hours of steady work faced us.
  
  
  NOTE {"id": "5c7299fd-a233-bdc0-9d63-865f9a4abd75", "timestamp": "2025-10-29T18:32:06.553163+00:00"}
  
  5c7299fd-a233-bdc0-9d63-865f9a4abd75
  00:00:29.800 --> 00:00:32.440
  A large size in stockings is hard to sell.
  
  '''
# ---
# name: test_transcribe_osr_audio[openai/whisper-tiny]
  '''
  WEBVTT
  
  NOTE {"id": "2ea43707-088b-c4fe-c7ff-b59f4a1232a0", "mediaFilePath": "OSR_us_000_0010_8k.wav"}
  
  
  NOTE {"id": "c10f5604-16d0-7a49-0fb8-851bbe52bc8f", "timestamp": "2025-10-29T18:39:11.512137+00:00"}
  
  c10f5604-16d0-7a49-0fb8-851bbe52bc8f
  00:00:00.000 --> 00:00:04.000
  The birch can use lid on the smooth planks.
  
  
  NOTE {"id": "57c15f3a-9d2c-f253-abdc-2554cec2381d", "timestamp": "2025-10-29T18:39:11.512137+00:00"}
  
  57c15f3a-9d2c-f253-abdc-2554cec2381d
  00:00:04.000 --> 00:00:07.000
  Glue the sheet to the dark blue background.
  
  
  NOTE {"id": "fd79e577-4f48-44ed-3cc2-cec43ffe3183", "timestamp": "2025-10-29T18:39:11.512137+00:00"}
  
  fd79e577-4f48-44ed-3cc2-cec43ffe3183
  00:00:07.000 --> 00:00:10.000
  It is easy to tell the depth of a well.
  
  
  NOTE {"id": "af1751b8-849a-9b18-c71c-1623e9f8c078", "timestamp": "2025-10-29T18:39:11.512137+00:00"}
  
  af1751b8-849a-9b18-c71c-1623e9f8c078
  00:00:11.800 --> 00:00:15.000
  make it a rare dish, rice is often
  
  
  NOTE {"id": "6cd980d5-26d5-6b85-1677-eb0055cfcadd", "timestamp": "2025-10-29T18:39:11.512137+00:00"}
  
  6cd980d5-26d5-6b85-1677-eb0055cfcadd
  00:00:17.760 --> 00:00:19.280
  of lemons makes fine punch.
  
  
  NOTE {"id": "ebf3daf0-cddb-b350-d2aa-de57b084c0e4", "timestamp": "2025-10-29T18:39:11.512137+00:00"}
  
  ebf3daf0-cddb-b350-d2aa-de57b084c0e4
  00:00:21.720 --> 00:00:23.720
  the puncture.
  
  
  NOTE {"id": "945da4c3-3a6e-3260-9734-6599082dda61", "timestamp": "2025-10-29T18:39:11.512137+00:00"}
  
  945da4c3-3a6e-3260-9734-6599082dda61
  00:00:23.720 --> 00:00:25.720
  The haves are fed chocolate.
  
  
  NOTE {"id": "46c4d02a-ba25-563a-a7de-ed6525ba40c7", "timestamp": "2025-10-29T18:39:11.512137+00:00"}
  
  46c4d02a-ba25-563a-a7de-ed6525ba40c7
  00:00:27.000 --> 00:00:29.000
  Four hours of steady work faced us.
  
  '''
# ---
